mlflow/gateway/providers/togetherai.py

```diff
@@ -0,0 +1,470 @@
+        #b'data: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":" of","logprobs":null,"finish_reason":null,"delta":{"token_id":302,"content":" of"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\ndata: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":" Living","logprobs":null,"finish_reason":null,"delta":{"token_id":16303,"content":" Living"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\ndata: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":" Survey","logprobs":null,"finish_reason":null,"delta":{"token_id":24004,"content":" Survey"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\ndata: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":" in","logprobs":null,"finish_reason":null,"delta":{"token_id":297,"content":" in"}}],"model":"mistralai/Mixtral-8x'
+        #b'7B-v0.1","usage":null}\n\n'
+        #b'data: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":" ","logprobs":null,"finish_reason":null,"delta":{"token_id":28705,"content":" "}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\ndata: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":"2","logprobs":null,"finish_reason":null,"delta":{"token_id":28750,"content":"2"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":null}\n\ndata: {"id":"86dfe9397f84eed4-ATH","object":"chat.completion.chunk","created":1712051391,"choices":[{"index":0,"text":"0","logprobs":null,"finish_reason":"length","delta":{"token_id":28734,"content":"0"}}],"model":"mistralai/Mixtral-8x7B-v0.1","usage":{"prompt_tokens":93,"completion_tokens":200,"total_tokens":293}}\n\n'
+        #b'data: [DONE]\n\n'
+
+        return chat.StreamResponsePayload(
+            id=resp["id"],
+            model=config.model.name,
+            object="chat.completion.chunk",
+            created=resp["created"],
+            choices=[
+                chat.StreamChoice(
+                    index=idx,
+                    # TODO: is this a good idea? This comes from the TogetherAI API
+                    finish_reason=choice.get("finish_reason"),
+                    delta=chat.StreamDelta(
+                        role=None,
+                        content=choice.get("text"),
+                    )
+                )
+                # Added enumerate and a default empty list
+                for idx, choice in enumerate(resp.get("choices", []))
+            ],
+            usage=resp.get("usage")
+        )
+
+    @classmethod
+    def chat_to_model(cls, payload, config):
+
+        if "prompt" in payload:
```

Comment:
Same as above, we don't do client-side validation and the underlying API should raise the error.